### ISSUE TITLE
shadow position

### ISSUE THREAD

**Issue by kberkut** [yshui/picom#293]
1) Shadow position not work for zero coordinates:
- The left offset for shadows. (default -15)
shadow-offset-x = 0;
 - The top offset for shadows. (default -15)
shadow-offset-y = 0;
![2020-01-09_21-09](https://user-images.githubusercontent.com/57869227/72093968-76047980-3326-11ea-88ed-a70dc089f053.png)

2) How to make shadow visible only for active windows?

### Platform
OS: Arch Linux x86_64 
Kernel: 5.4.8-arch1-1

### Environment
WM: i3

### picom --diagnostics                                     
**Version:** v7.5

### Extensions:
* Shape: Yes
* XRandR: Yes
* Present: Present

### Misc:
* Use Overlay: Yes
* Config file used: /home/dima/.config/picom.conf

### Drivers (inaccurate):
AMDGPU, Radeon

**Comment by kberkut** [comment]
If Shadow-radius=0 then position shadow same as coordinates.

![2020-01-10_05-45](https://user-images.githubusercontent.com/57869227/72121849-4dec3900-336d-11ea-86d8-68f1e6e2968b.png)
![2020-01-10_05-47](https://user-images.githubusercontent.com/57869227/72121859-52185680-336d-11ea-8146-0533ff6209f2.png)

If Shadow-radius not equil zero then position shadow not same as coordinates.
![2020-01-09_21-09](https://user-images.githubusercontent.com/57869227/72121999-ac191c00-336d-11ea-86fd-e6d1aefffcf5.png)
![2020-01-10_05-48](https://user-images.githubusercontent.com/57869227/72121865-55abdd80-336d-11ea-8715-aed0e142a8a2.png)

**Comment by mikeroll** [comment]
1. It's a bit hard to follow, but my guess is you want the shadow to poke out evenly from each of the four window sides. Set `shadow-offset-{x,y}` to `- shadow-radius`, e.g.:
```
shadow-radius = 15
shadow-offset-x = -15
shadow-offset-y = -15
```
When changing the radius, also adjust the other two.

2. Try adding `'!focused'` rule to your `shadow-exclude`:
```
shadow-exclude = [
    ...
    '!focused',
]
```

**Comment by kberkut** [comment]
Then adding 
shadow-exclude = [
    ...
    **'!focused',**
]

I have error
 
`Error when reading configuration file "/home/user/.config/picom.conf", line 99: syntax error
[ 17.01.2020 16:51:45.463 main FATAL ERROR ] Failed to create new session.
`

**Comment by yshui** [comment]
@kberkut 

1. no matter what is shadow radius is, the top left corner of the shadow always align with the top left corner of the window. so this is working as expected. 

2. @mikeroll meant adding `'!focused'` to your `shadow-exclude` array, you shouldn't put the ellipsis (`...`) there too.

**Comment by yshui** [comment]
Closing as this is not a bug.

**Comment by cor** [comment]
I'm having the exact same issue. It is preventing me from creating macOS-like shadows.



> 1. no matter what is shadow radius is, the top left corner of the shadow always align with the top left corner of the window. so this is working as expected

@yshui why should the top left corner of the window always be aligned with the top left corner of the shadow? This is quite limiting

**Comment by yshui** [comment]
@cor did you try the suggestion in https://github.com/yshui/picom/issues/293#issuecomment-575627428
